src/robot-handler.cpp

```diff
@@ -11,37 +11,19 @@ namespace simple_mpc
 {
 
   RobotModelHandler::RobotModelHandler(
-    const Model & model,
-    const std::string & reference_configuration_name,
-    const std::string & base_frame_name,
-    const std::vector<std::string> & locked_joint_names)
-  : model_full_(model)
+    const Model & model, const std::string & reference_configuration_name, const std::string & base_frame_name)
+  : model_(model)
   {
-    // Construct controlled and locked joints ids list
-    std::vector<unsigned long> locked_joint_ids;
-    for (size_t i = 1; i < model_full_.names.size(); i++)
-    {
-      const std::string joint_name = model_full_.names.at(i);
-      if (count(locked_joint_names.begin(), locked_joint_names.end(), joint_name) == 0)
-      {
-        controlled_joints_ids_.push_back(model_full_.getJointId(joint_name));
-      }
-      else
-      {
-        locked_joint_ids.push_back(model_full_.getJointId(joint_name));
-      }
-    }
-
-    // Build reduced model with locked joints
-    buildReducedModel(
-      model_full_, locked_joint_ids, model_full_.referenceConfigurations[reference_configuration_name], model_);
+    // Controlled joints index
+    for (auto joint_name : model_.names)
+      controlled_joints_ids_.push_back(model.getJointId(joint_name));
```

Comment:
I think the `controlled_joint_ids` can be deleted too, since there's no more full model with certain joints controlled. All the joints on the model are now controlled